src/umpire/strategy/mixins/AlignedAllocation.inl

```diff
@@ -17,28 +17,30 @@ namespace mixins {
 
 inline std::size_t AlignedAllocation::aligned_round_up(std::size_t size)
 {
-    return size + (m_alignment - 1) - (size - 1) % m_alignment;
+  return size + (m_alignment - 1) - (size - 1) % m_alignment;
 }
 
 inline void* AlignedAllocation::aligned_allocate(std::size_t size)
 {
-    std::size_t total_bytes{ size + m_alignment - 1 };
-    uintptr_t ptr{ reinterpret_cast<uintptr_t>(m_allocator->allocate(total_bytes)) };
+  std::size_t total_bytes{ size + m_alignment };
+  uintptr_t ptr{ reinterpret_cast<uintptr_t>(m_allocator->allocate(total_bytes)) };
```

Comment:
Before doing this, why don't you check if the underlying allocator already produces aligned addresses? Because of this, when using CUDA, and despite our 256-bytes aligned memory guarantee (c.f. [CUDA programming guide](https://docs.nvidia.com/cuda/cuda-c-programming-guide/index.html#device-memory-accesses), section 5.3.2.1.1), you might allocate too much memory (e.g. if size == 4096, you will allocate 4096 + 16 == 4112).

This might be a large change considering the way you stack allocation strategies, don't know if it's really worth...

Reply:
This is a fair question.  This mixin is intended for pool strategies.  I wasn't concerned about the additional size since this allocation will be made for large chunks of memory (several megabytes to gigabytes range) to be added to the pool.

Reply:
Since applications may ask for arbitrary alignments ranging from a small power of 2 to potentially page-size multiples, this is the mechanism we are using which is wasteful as noted above, but is small compared relative to the sizes being allocated by the pool which is in the multi-MiB/GiB range.  Let me know if this explanation sounds reasonable and I will mark this conversation as resolved.